src/components/Layout/MoviesList/MoviesList.tsx

```diff
@@ -1,16 +1,19 @@
-import DATA from '../../../constants/data.json';
-import { MovieItem, MovieItemProps } from '../MovieItem/MovieItem';
+import { useAppSelector } from '../../../redux/store';
+import { api } from '../../../services/api';
+import { MovieItem } from '../MovieItem/MovieItem';
 
 import './MoviesList.scss';
 
 interface MoviesListProps {
   searchEntry: string;
 }
 const MoviesList = (_props: MoviesListProps): JSX.Element => {
-  const movies: MovieItemProps[] = DATA.data;
+  const sortBy = useAppSelector((state) => state.movies.sortBy);
+  const filters = useAppSelector((state) => state.movies.filters);
+  const { data = [] } = api.useGetMoviesQuery({ sortBy, filters });
 
-  if (movies) {
-    const card = movies.map((movie) => <MovieItem {...movie} key={movie.id} />);
+  if (data) {
```

Comment:
You need to check for `.length`, I believe, otherwise the default value of `[]` will always be truthful.

Reply:
Yes, you are right.